### ISSUE TITLE
Asset server

### ISSUE THREAD

**Issue by bob-ludvigsen** [adlnet/Sandbox#287]
I tried to upload the attached hangar to the asset server, but the spinner just goes on until the server connection is lost. I tried a couple of files, but no luck..

[hangar.zip](https://github.com/adlnet/Sandbox/files/87814/hangar.zip)

**Comment by rchadwic** [comment]
Thanks for bringing that up - I saw some logs that indicated a problem. The 3DR (which is really a different project altogether) is down in an interesting way, and returning odd codes. We need to handle the responses from that server better. I've made a commit to fail more gracefully in this case. We also need to fix the 3DR, but that is another problem.

**Comment by lazydino** [comment]
Is 3DR still working? I recall there was a web page interface that allows you to login and manage downloads/uploads. Is that web application still running?

**Comment by rchadwic** [comment]
The 3DR is no longer hosted by ADL, but we do still make the software available. You can find an Amazon Machine Image (AMI) in the AWS Instance Store that contains a working copy of the 3DR. The ID for that machine is ami-d2e2cfb8.

It should also come with a working install of Sandbox( though an old one.) You can look at the configuration file in that Sandbox install to see how you can hook up other Sandbox installs to your own 3DR.

**Comment by lazydino** [comment]
Is the asset management in the current code utilizing 3DR Sandbox asset server at https://github.com/adlnet/SandboxAssetServer ? Is it used as a node module or independent server?

**Comment by rchadwic** [comment]
Yes - it is installed by default as a module, and the default configuration hosts the service within the same process as the main Sandbox server. 

For a more advanced setup (one with more than a single Sandbox server) it's necessary to host the asset server independently. In most cases, you shouldn't have to do anything, and the defaults should work.
